pkg/services/hcloud/server/server.go

```diff
@@ -925,3 +1432,67 @@ func updateHCloudMachineStatusFromServer(hm *infrav1.HCloudMachine, server *hclo
 	hm.Status.Addresses = statusAddresses(server)
 	hm.Status.InstanceState = ptr.To(server.Status)
 }
+
+func (s *Service) getSSHClient(ctx context.Context) (sshclient.Client, error) {
```

Comment:
add a description to this function. It takes the robot ssh secret - this is a bit strange and should be mentioned